packages/mui-material/src/Paper/Paper.js

```diff
@@ -107,6 +113,22 @@ const Paper = React.forwardRef(function Paper(inProps, ref) {
       className={clsx(classes.root, className)}
       ref={ref}
       {...other}
+      style={{
+        ...(variant === 'elevation' && {
+          '--Paper-shadow': (theme.vars || theme).shadows[elevation],
```

Comment:
We should prefix this variable no? `--Paper-shadow` feels like one that would easily conflict. If we match the values in the theme

```suggestion
          '--MuiPaper-shadow': (theme.vars || theme).shadows[elevation],
```